ext/BeamTrackingBeamlinesExt/integration.jl

```diff
@@ -0,0 +1,171 @@
+# =========== HELPER FUNCTIONS ============= #
+@inline function integration_launcher!(ker, params, tm, L)
+  order = tm.order
+  ds_step = tm.ds_step
+  num_steps = tm.num_steps
+  if ds_step < 0
+    ds_step = L / num_steps
+  else
+    num_steps = Int(ceil(L / ds_step))
+    ds_step = L / num_steps
+  end
+  if order == 2
+    return KernelCall(IntegrationTracking.order_two_integrator!, (ker, params, ds_step, num_steps, L))
+  elseif order == 4
+    return KernelCall(IntegrationTracking.order_four_integrator!, (ker, params, ds_step, num_steps, L))
+  elseif order == 6
+    return KernelCall(IntegrationTracking.order_six_integrator!, (ker, params, ds_step, num_steps, L))
+  elseif order == 8
+    return KernelCall(IntegrationTracking.order_eight_integrator!, (ker, params, ds_step, num_steps, L))
+  end
+end
+
+
+# =========== STRAIGHT ELEMENTS ============= #
+# === Thin elements === #
+@inline function thin_pure_bdipole(tm::SplitIntegration, bunch, bm)
+  brho_0 = bunch.Brho_ref
+  mm = bm.order
+  knl, ksl = get_integrated_strengths(bm, 0, brho_0)
+  return KernelCall(ExactTracking.multipole_kick!, (mm, knl, ksl, 1, 1))
+end
+
+@inline thin_bdipole(tm::SplitIntegration, bunch, bm) = thin_pure_bdipole(tm, bunch, bm)
+
+@inline thin_pure_bquadrupole(tm::SplitIntegration, bunch, bm) = thin_pure_bdipole(tm, bunch, bm)
+
+@inline thin_bquadrupole(tm::SplitIntegration, bunch, bm) = thin_pure_bdipole(tm, bunch, bm)
+
+@inline thin_pure_bmultipole(tm::SplitIntegration, bunch, bm) = thin_pure_bdipole(tm, bunch, bm)
+
+@inline thin_bmultipole(tm::SplitIntegration, bunch, bm) = thin_pure_bdipole(tm, bunch, bm)
+
+
+# === Thick elements === #
+@inline drift(tm::Union{SplitIntegration,DriftKick}, bunch, L) = drift(Exact(), bunch, L)
+
+@inline thick_pure_bsolenoid(tm::Union{SplitIntegration,SolenoidKick}, bunch, bm, L) = 
+  thick_pure_bsolenoid(Exact(), bunch, bm, L)
+
+@inline function thick_bsolenoid(tm::Union{SplitIntegration,SolenoidKick}, bunch, bm, L) 
+  brho_0 = bunch.Brho_ref
+  tilde_m, gamsqr_0, beta_0 = ExactTracking.drift_params(bunch.species, brho_0)
+  mm = bm.order
+  kn, ks = get_strengths(bm, L, brho_0)
+  Ksol = sqrt(kn[1]^2 + ks[1]^2)
+  params = (beta_0, gamsqr_0, tilde_m, Ksol, mm, kn, ks)
+  return integration_launcher!(IntegrationTracking.sks_multipole!, params, tm, L)
+end
+
+@inline function thick_pure_bdipole(tm::Union{SplitIntegration,DriftKick}, bunch, bm, L)
+  brho_0 = bunch.Brho_ref
+  tilde_m, gamsqr_0, beta_0 = ExactTracking.drift_params(bunch.species, brho_0)
+  mm = bm.order
+  kn, ks = get_strengths(bm, L, brho_0)
```

Comment:
```suggestion
  knt, kst = get_strengths(bm, L, brho_0)
  kn = SVector(knt)
  ks = SVector(kst)
```

Instead of multiplying L inside the kernel, let's just make `kn` and `ks` be `SVector`s (stack allocated) instead of `SizedVector`s.